service-config/src/main/java/com/palantir/config/service/ServiceConfiguration.java

```diff
@@ -43,4 +43,23 @@
      * A list of service URIs.
      */
     public abstract List<String> uris();
+
+    // hides implementation details
+    public static Builder builder() {
+        return ImmutableServiceConfiguration.builder();
+    }
+
+    // hides implementation details
+    public interface Builder {
+
+        Builder apiToken(BearerToken apiToken);
+
+        Builder security(SslConfiguration security);
```

Comment:
I didn't notice this until I tried using it - this should also have
Builder security(Optional<SslConfiguration> security)
so we can directly pass one variable to another. Immutables does this for you but by making your own interface you bypass the auto-generation.

Can we please just not have this "hide implementation interface" class?